docs/src/dimarrays.md

```diff
@@ -58,6 +58,13 @@ fill(7, X(5), Y(10))
 fill(7, X(5), Y(10); name=:fill, metadata=Dict())
 ```
 
+== generator expression
+
+```@ansi dimarray
+DimArray((x, y) for x in X(10:10:50), y in Y(0.0:0.1:1.0))
```

Comment:
```suggestion
((x, y) for x in X(10:10:50), y in Y(0.0:0.1:1.0))
```

This should work too?

Reply:
Good to show the simplest version first

Reply:
Ah yes, good point.
I guess I was in the mindset of getting keyword arguments working with a generator constructor.

Reply:
Shall I add an array comprehension syntax example here too?

Reply:
Oh yeah that's what I meant sorry

Reply:
It works with both actually. The square brackets creates an array, but the round brackets creates a generator.

That said, I should remove the round brackets then, I realise as I type this. 😅